lib/api-base/authorizer.ts

```diff
@@ -82,12 +86,16 @@ export class CustomAuthorizer extends Construct {
                 AUTHORITY: config.authConfig!.authority,
                 ADMIN_GROUP: config.authConfig!.adminGroup,
                 JWT_GROUPS_PROP: config.authConfig!.jwtGroupsProperty,
+                MANAGEMENT_KEY_NAME: managementKeySecretNameStringParameter.stringValue
             },
             role: role,
             vpc: vpc,
             securityGroups: securityGroups,
         });
 
+        const managementKeySecret = Secret.fromSecretNameV2(this, createCdkId([id, 'managementKey']), managementKeySecretNameStringParameter.stringValue);
+        managementKeySecret.grantRead(authorizerLambda);
```

Comment:
is this correct? or should the read be granted to the authorizer lambda role? i normally would have expected that the execution role gets the permissions

Reply:
It does add it to the lambda role. I can try adding it for the role directly though.

Reply:
Actually since `role` is nullable and it works like this it is probably best to keep it as is since we know that the `authorizerLambda` will never be null.

Reply:
TIL. nice!